plugins/link/SharedMemory.h

```diff
@@ -22,17 +24,22 @@ class SharedMemory {
 	explicit SharedMemory();
 	~SharedMemory();
 
-	std::size_t size();
-
 	void close();
 
 	int lastError() const;
 
-	void *mapMemory(const char *name, std::size_t size);
+	bool mapMemory(const char *name);
+
+	bool isMemoryMapped();
+
+	void read(LinkedMem *dest);
+
+	void write(const LinkedMem *source);
```

Comment:
These should take a reference rather than a pointer as passing `nullptr` to either of them clearly is not a valid use case.
```suggestion
	void read(LinkedMem &dest);

	void write(const LinkedMem &source);
```